types/run.sh

```diff
@@ -94,6 +94,37 @@ typecheck-files() {
   $0 typecheck --follow-imports=silent $MYPY_FLAGS "$@"
 }
 
+readonly -a COMMON_TYPE_MODULES=(_devbuild/gen/runtime_asdl.py _devbuild/gen/syntax_asdl.py)
+
+add-imports() {
+	# Temporary helper to add missing class imports to the 'if
+	# TYPE_CHECKING:' block of a single module, if the relevant
+	# classes are found in one of COMMON_TYPE_MODULES
+
+	# Also, this prints out the typechecking output, to avoid having
+	# to run two redundant (and slow) typechecking commands.
+	local module=$1
+	export PYTHONPATH=.
+	module_tmp=$(mktemp)
```

Comment:
- I find it easier to read and debug to use a hard-coded well-named path in `_tmp`.  Then if something goes wrong you can look at those files manually.
- And I wouldn't bother with `trap` because of this.
- please follow the existing style of using two spaces for indents here

Reply:
@andychu sorry about the indentation, I didn't notice!  I've installed an emacs mode that I expect to make this never happen again (for shell, at least -- it doesn't do python).  Also fixed the need-typechecking function I added.  dde3ffb02ae0c08101bb47a712b1f8578ab3fe18

Temporary file paths have been sorted as well.  92d8805ba3649e6464f999f4b3a57941cc7e165c

Reply:
Oh, and one more change -- f48392b8f85596ae1db9a356ba0af76e590df079.  I thought it would be simpler to just save the mypy output and let the caller can decide whether to print it or not.